src/client/components/layout.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2016  Max Prettyjohns
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+const React = require('react');
+
+const bootstrap = require('react-bootstrap');
+
+const Nav = bootstrap.Nav;
+const Navbar = bootstrap.Navbar;
+const MenuItem = bootstrap.MenuItem;
+const FormGroup = bootstrap.FormGroup;
+
+const Footer = require('./footer');
+
+
+/**
+ * Returns list of component children that have been injected with the specified props (does not override existing ones)
+ * @param {Object} props - The props object that contains children and will be re-injected into children
+ * @returns {Array} list of children
+ */
+function injectChildElemsWithProps(props) {
+	'use strict';
+	return React.Children.map(props.children, (Child) => {
+		const filteredProps = Object.assign({}, props, Child.props);
+		return React.cloneElement(Child, filteredProps);
+	});
+}
+
+class Layout extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {};
+		this.renderNavContent = this.renderNavContent.bind(this);
+		this.renderNavHeader = this.renderNavHeader.bind(this);
+	}
+
+	renderNavHeader() {
+		const {homepage} = this.props;
+
+		return (
+			<Navbar.Header>
+				<Navbar.Brand className="logo" href="/">
+					{homepage ?
+						<img alt="BookBrainz icon" src="/images/BookBrainz_logo_icon.svg" title="BookBrainz"/> :
+						<img alt="BookBrainz icon" src="/images/BookBrainz_logo_mini.svg" title="BookBrainz"/>
+					}
+				</Navbar.Brand>
+				<Navbar.Toggle/>
+			</Navbar.Header>
+		);
+	}
+
+	renderNavContent() {
+		const {user, homepage, hideSearch} = this.props;
+
+		return (
+			<Navbar.Collapse id="bs-example-navbar-collapse-1">
+				{user && user.id ?
+					<Nav pullRight>
+						<li className="dropdown">
+							<a aria-expanded="false" className="dropdown-toggle" data-toggle="dropdown" href="#" id="dNewEntities" role="button">
+								<span className="fa fa-plus"/>&nbsp;Create&nbsp;
```

Comment:
Can simplify this by importing font-awesome React components (react-fontawesome).